esp/xslt/esp_service_xml2xsd.xslt

```diff
@@ -159,11 +160,15 @@
                     </xsl:when>
                 </xsl:choose>
             </xsl:attribute>
+
             <!--
-                There are some .ecm files which use 'boolean' instead of 'bool', and for some reason those fields
-                don't ever have a 'default' attribute when the XSD is generated. Reproduce that behavior here:
+                When the XSD was generated by HIDL code, int, integer, int64, and long types would not output a
+                default="0" attribute if a default of 0 was specified. It is more correct to have the default
+                attribute present in all cases when a default is given. To maintain compatibility, the
+                output_zero_default_values parameter can be used to create an XSD w/o a default attribute
+                for these types. The parameter default is set above.
              -->
-            <xsl:if test="@default and @type != 'boolean'">
+            <xsl:if test="@default and ($output_zero_default_values or (@default!='0' or (@type!='int' and @type!='integer' and @type!='int64' and @type!='long')))">
```

Comment:
Does this maintain the same behavior regarding 'boolean' elements? I see that there are some with default values in ws_sql.ecm.

Reply:
@asselitx The comment about boolean behavior was not accurate. I checked for all instances of boolean in the SCM files and for each that had a default value, the HIDL generated schema also had a default value in their output. Therefore I removed that test to ensure that boolean values with a default would have a default attribute in their schema

Reply:
given that, looks good!